client/state/sites/selectors.js

```diff
@@ -281,6 +283,24 @@ export function isSitePreviewable( state, siteId ) {
 }
 
 /**
+ * Returns true if the site can be upgraded by the user, false if the
+ * site cannot be upgraded, or null if upgrade ability cannot be
+ * determined.
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether site is upgradeable
+ */
+export function isSiteUpgradeable( state, siteId ) {
+	// Cannot determine site upgradeability if there is no current user
+	if ( ! getCurrentUserId( state ) || ! getRawSite( state, siteId ) ) {
```

Comment:
For my own understanding, why is `getRawSite` check required?

As I understand, we suppose to check: 
`canCurrentUser(...)` which is `state.currentUser.capabilities[ siteId ][ capability ]`. 
that calls `isValidCapability` which returns `null` if `! state.currentUser.capabilities[ siteId ]`,
so that seems already be covered...